workflow/controller/operator.go

```diff
@@ -1199,6 +1190,29 @@ func (woc *wfOperationCtx) assessNodeStatus(pod *apiv1.Pod, node *wfv1.NodeStatu
 	return nil
 }
 
+func getPodTemplate(pod *apiv1.Pod) (*wfv1.Template, error) {
+	tmpl := &wfv1.Template{}
+	for _, c := range pod.Spec.Containers {
+		for _, e := range c.Env {
+			if e.Name == common.EnvVarTemplate {
+				return tmpl, json.Unmarshal([]byte(e.Value), tmpl)
+			}
+		}
+	}
+	return nil, fmt.Errorf("not found")
+}
+
+func getPodDeadline(pod *apiv1.Pod) (time.Time, error) {
+	for _, c := range pod.Spec.Containers {
+		for _, e := range c.Env {
+			if e.Name == common.EnvVarDeadline {
+				return time.Parse(time.RFC3339, e.Value)
```

Comment:
As above. Old pods won’t have this variable. Instead, can we robustly determine it based on workflow? I suspect not, as it is determined from some opts. If not, then when invoking this func, we should ignore errors (probably log them).

Reply:
Same as above. The `getPodDeadline()` has been moved to `operator_test.go`, since it is only used by tests.